plugin/cassandra/plugin.go

```diff
@@ -90,106 +90,151 @@ package main
 		Author:  "Orange",
-		Version: "0.1.0",
+		Version: "1.0.0",
 		Features: plugin.PluginFeatures{
 			Target: "yes",
 			Store:  "no",
 		},
 		Example: `
 {
-  "host"         : "127.0.0.1",      # optional
-  "port"         : "9042",           # optional
-  "user"         : "username",
-  "password"     : "password",
-  "keyspace"     : "db",
-  "datadir"      : "/path/to/data"   # optional
+  "host"              : "127.0.0.1",      # optional
+  "port"              : "9042",           # optional
+  "user"              : "username",
+  "password"          : "password",
+  "include_keyspaces" : ["db"],
+  "exclude_keyspaces" : ["system"],
+  "save_users"        : true,
+  "bindir"            : "/path/to/bin",   # optional
+  "datadir"           : "/path/to/data",  # optional
 }
 `,
 		Defaults: `
 {
-  "host"     : "127.0.0.1",
-  "port"     : "9042",
-  "user"     : "cassandra",
-  "password" : "cassandra",
-  "datadir"  : "/var/vcap/store/cassandra/data"
+  "host"              : "cassandra_host", // "12.0.0.1"
+  "port"              : "9042", // "cassandra_port"
+  "user"              : "cassandra_user",     //cassandra
+  "password"          : "cassandra_password", // "cassandra"
+  "include_keyspaces" : [""],
+  "exclude_keyspaces" : [ "system_schema", "system_distributed", "system_auth", "system", "system_traces" ],
+  "save_users"        : true,
+  "bindir"            : "/var/vcap/jobs/cassandra/bin", 
+  "datadir"           : "/var/vcap/store/cassandra/data" 
 }
 `,
 		Fields: []plugin.Field{
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "host",
 				Type:     "string",
 				Title:    "Cassandra Host",
 				Help:     "The hostname or IP address of your Cassandra server.",
 				Default:  "127.0.0.1",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "port",
 				Type:     "port",
 				Title:    "Cassandra Port",
 				Help:     "The 'native transport' TCP port that Cassandra server is bound to, listening for incoming connections.",
 				Default:  "9042",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "user",
 				Type:     "string",
 				Title:    "Cassandra Username",
 				Help:     "Username to authenticate to Cassandra as.",
 				Default:  "cassandra",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
-				Name:     "password",
+				Name:     "cassandra",
 				Type:     "password",
 				Title:    "Cassandra Password",
-				Help:     "Password to authenticate to Cassandra as.",
+				Help:     "Password to auth_userenv",
 				Default:  "cassandra",
-				Required: true,
+				Required: false,
 			},
-			plugin.Field{
+			{
 				Mode:     "target",
-				Name:     "keyspace",
-				Type:     "string",
-				Title:    "Keyspace to Backup",
-				Help:     "The name of the keyspace to backup.",
-				Example:  "system",
-				Required: true,
```

Comment:
For backwards-compatibility, we need to retain the `keyspace` field, but make it optional.  It should probably be mutually-exclusive with the new `include_keyspaces` / `exclude_keyspaces` options.  It would seem that a setting of `keyspace` should be interpreted as a setting of `include_keyspaces` containing a single element, the value.